components/search_results/search_limits_banner.tsx

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import styled from 'styled-components';
+import {useIntl} from 'react-intl';
+import {useSelector} from 'react-redux';
+
+import useOpenPricingModal from 'components/common/hooks/useOpenPricingModal';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {isAdmin} from 'mattermost-redux/utils/user_utils';
+import {isCurrentLicenseCloud, getSubscriptionProduct as selectSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {CloudProducts} from 'utils/constants';
+import useGetUsage from 'components/common/hooks/useGetUsage';
+import useGetLimits from 'components/common/hooks/useGetLimits';
+import {asGBString, fallbackStarterLimits} from 'utils/limits';
+
+const StyledDiv = styled.div`
+width: 100%;
+`;
+
+const StyledA = styled.a`
+color: var(--denim-button-bg) !important;
+`;
+
+const InnerDiv = styled.div`
+display: flex;
+gap: 8px;
+border: 1px solid rgba(var(--center-channel-text-rgb), 0.08);
+border-radius: 4px;
+background-color: rgba(var(--center-channel-text-rgb), 0.04);
+padding: 10px;
+margin: 10px;
+color: rgba(var(--center-channel-text-rgb), 0.72);
+font-weight: 400;
+font-size: 11px;
+line-height: 16px;
+letter-spacing: 0.02em;
+`;
+
+type Props = {
+    searchType: string;
+}
+
+const FILES_SEARCH_TYPE = 'files';
+const MESSAGES_SEARCH_TYPE = 'messages';
+
+function SearchLimitsBanner(props: Props) {
+    const {formatMessage, formatNumber} = useIntl();
+    const openPricingModal = useOpenPricingModal();
+    const usage = useGetUsage();
+    const [cloudLimits] = useGetLimits();
+    const isAdminUser = isAdmin(useSelector(getCurrentUser).roles);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const product = useSelector(selectSubscriptionProduct);
+    const isStarter = product?.sku === CloudProducts.STARTER;
+
+    if (!isCloud || !isStarter) {
```

Comment:
If the subscription is in Professional they do have limits on Files. I believe we can just rely on the actual limits on the subscription and here just return if it's cloud
```suggestion
    if (!isCloud) {
```